Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -675,8 +676,9 @@ class Navigator {
                        state.navigation.navSurfaceIndex)
                    << " out of " << state.navigation.navSurfaces.size()
                    << " surfaces remain to try.");
+      const auto& intersection = state.navigation.navSurface();
       // Take the surface
-      auto surface = state.navigation.navSurface().object();
+      auto surface = intersection.object();
```

Comment:
```suggestion
      const auto* surface = intersection.object();
```
to make the pointer type explicit.